extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/QuarkusErrorServlet.java

```diff
@@ -35,9 +35,20 @@ protected void service(HttpServletRequest req, HttpServletResponse resp) throws
             details += "Error id " + uuid;
         }
 
-        resp.setContentType("text/html");
-        resp.setCharacterEncoding(StandardCharsets.UTF_8.name());
-        resp.getWriter().write(new TemplateHtmlBuilder().error(details).stack(stack).toString());
+        String accept = req.getHeader("Accept");
+        if (accept != null && accept.contains("application/json")) {
```

Comment:
It would be nice to see if we can use some content negotiation code in Undertow. I don't know if there is any though.

Reply:
I cannot find any content negociation in undertow so if needed I will need to build my own. And content negociation is complex (need to leverage the weight of each part of the Accept header). 

Here we favor json each time json is part of the Accept header, else we generates HTML message. This is a very _opiniated_ content negociation but a server can choose it's own algorithm and use the Accept header _as a hint_ only it doesn't break the HTTP spec.

I can implement a better alogithm if needed. But will this PR be merged some days or need it to be build on top of Vert.x ?

Maybe a broader PR should be created to cover all error pages (resteasy, Vert.x, undertow, security, validation) for multiple accept header (HTML, json, text) ?

Reply:
done